lib/HLSL/HLOperationLower.cpp

```diff
@@ -6485,7 +6553,8 @@ void TranslateStructBufSubscriptUser(Instruction *user, Value *handle,
       TranslateStructBufMatLdSt(userCall, handle, OP, status, bufIdx,
                                 baseOffset, DL);
     else if (group == HLOpcodeGroup::HLSubscript) {
-      TranslateStructBufMatSubscript(userCall, handle, OP, bufIdx, baseOffset, status, DL);
+      TranslateStructBufMatSubscript(userCall,
+        handle, ResKind, bufIdx, baseOffset, status, OP, DL);
```

Comment:
`OP`'s position changed to second last, whereas in other places (such as `TranslateStructBufMatLdSt`), it appears immediately after `handle`. was this intentional?

Reply:
I actually made it consistent with `TranslateStructBufSubscriptUser`, but the whole file is a mess. I guess the new state isn't worse than the old one?

Reply:
hmm, yes, this could cleaned up later. Not a high-pri for now, imo..